cmd/piper.go

```diff
@@ -1,5 +1,7 @@
 package cmd
 
+//go:generate go run ../pkg/generator/step-metadata.go --metadataDir=../resources/metadata/
```

Comment:
../ is a bit ugly. I assume we have to cd here because go generate expects this comment in pwd relative to its invocation, right?

Reply:
If so, we should move it to `main.go`.

Reply:
I can't judge the consequences of this, but it sounds good go me, if it works.

Reply:
the comment needs to be in a file in the correct package. I think we should not move the generator to our `cmd` package ...

Reply:
I cleaned up a bit ...
We need to keep in mind that a call to the same file also triggers doc generation.

Reply:
For me it worked also when moving the command to `main.go`.

Reply:
Thanks! I merged your PR into this one ...